### ISSUE TITLE
No enum_of_keycode ?

### ISSUE THREAD

**Issue by haesbaert** [dbuenzli/tsdl#67]
I've noticed there is no enum_of_keycode, there is enum_of_scancode though.

Any special reason ? I'm probably missing something but is there a way to match against keycode instead of using a huge string of if conditionals ?

**Comment by haesbaert** [comment]
> I may be wrong but I think that the proper way of solving that problem is to base your keyboard controller on scancodes and have an UI for your users that allows to redefine those according to their layout – and if you have the keyboard layout information set up a right default for them, maybe (?) SDL_GetScancodeFromKey can help with this ?

That would work, still I'm not convinced scancodes are the way to go. I'd prefer the user inputs the key the "way he knows" instead of having to remap his keyboard. I don't feel strongly about allowing users to rebind keys for now at least.

At any rate, for the sake of symmetry would it not make sense to add enum_of_keycode ?

**Comment by dbuenzli** [comment]
> I'd prefer the user inputs the key the "way he knows" instead of having to remap his keyboard

It's not really the "way he knows"; as demonstrated above it's rather the "way you think he knows"...  Again if you have good default which you might get by simply using the inverse map (I don't now, never tried), most users may not have to rebind keys. 

> At any rate, for the sake of symmetry would it not make sense to add enum_of_keycode ?

I think the notion of symmetry is a poor API design guide. My API design guide is to provide the elements that make programmers do the right thing for end-users. 

If you want to provide a patch I might consider it. But so far I'm not convinced by the utility of this; I also think your use case borderlines text-input events.

**Comment by haesbaert** [comment]
Yeah, not a big deal for me so I won't be sending a patch, thanks for the clarification. I think you should close this.

**Comment by dbuenzli** [comment]
I'll keep it open for now to see if the future brings us more dissenting opinions.

**Comment by jgerrish** [comment]
Just wanted to say thank you for the informative discussion in this thread around issues of language and enumeration.  I found the thread because some keys I was searching for (?) weren't enumerated in enum_of_scancode.

But I wasn't thinking in terms of the right use case. Physical keyboard layout is another axis as this thread points out and I learned some information.

Thank you for the project, it was still incredibly useful.  Have a great day!
